### ISSUE TITLE
Alerts Not Working

### ISSUE THREAD

**Issue by oslcit** [15thnight/15thnight#175]
Megan reported today that alerts were not being sent.  They would submit an alert and received the message at the top of the page "An unknown error occured while sending alert".  I rolled back the code to an older date and now when you submit an alert you receive the message "Alert sent successfully", but messages are not sent.  I tried to roll back the site to several different dates, including to the major release the night before 15th Night event, but same issue occurs.

I am viewing git events with command "git reflog --date=iso", then I rollback to a previous code version with something like "git reset --hard 11c10b7", where 11c10b7 is an example of a hash associated with a particular date.  Each time I use "git reset" I restart PostgreSQL, Apache and Celery.  Possibly this isn't the correct procedure?

When I use ./manage.py twillio_test I receive a text, so SMS is working.  I can't test alerts through the web interface though.  I can add, remove and edit users and I created a few test users.  However, when I select role type provider, select a service and submit, the message at the top of the page reports success, but the service is not added.  I can then edit the user and try to select a service again, but it doesn't work.  What I did instead was send an alert selecting the service "Dental Care", which Megan has a provider user account setup, and she is the only provider user account with this service selected.  When I send a alert for Dental Care the message at the top of the window reports "Alert sent successfully", but when I view Postfix logs I see no e-mails were sent.  I can see previous e-mails from Tuesday and older before the site was updated with "git pull".  I don't know if an SMS is sent because I don't know where or if that is logged.

The Celery daemon is running, and I've poked around to try to figure this out, but I've run out of ideas.  I'm hoping you may know the issue or can offer suggestions.  Any ideas?

**Comment by nobane** [comment]
I'm not immediately seeing the issue with the site. Let's try getting on the latest version, which is ready for use, and contains various fixes.

- Switch to the master branch and ensure it's up to date
```
$ git fetch origin
$ git checkout master
$ git rev-parse HEAD # should output da27c5705bdd905d7b2f3810048c452399784941
```
- Remove `HOST_NAME` from the `config.py` and add `SERVER_NAME` with no protocol, like so:
```
SERVER_NAME = '15thnight.help'
```
- Activate the virtual environment and run migrations
```
$ source venv/bin/activate
$ PYTHONPATH="." alembic upgrade head
```
- Update the front-end client
```
$ cd client
$ rm -rf node_modules
$ npm install
$ ./node_modules/.bin/webpack
$ cd ..
```
- Remove all .pyc files and restart Apache and Celery
```
$ find . -name \*.pyc -delete
$ # Restart the Apache and Celery daemons
```

I just ran through all of these steps on my test instance, [15thnight.spectralsun.com](15thnight.spectralsun.com). Everything appears to be working fine. I sent an alert through the whole system, and got all the correct text messages and emails. I was able to reset a password correctly. I was also able to create a provider from the admin form and select services.
